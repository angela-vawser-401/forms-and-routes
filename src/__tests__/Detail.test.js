import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetail from '../components/Detail';

jest.mock('../../services/avatarApi', () => ({
  getSingleCharacter() {
    return Promise.resolve([
      {
        '_id': '5da237699734fdcb7bef8f51',
        'name': 'Arnold Shortman',
        'photoUrl': 'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333',
        'gender': 'male'
      }
    ]);
  }
}));

describe('Character Detail component', () => {

  it('renders', () => {
    const wrapper = shallow(
      <CharacterDetail
        match={{
          params: {
            name: 'arnold'
          }
        }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
