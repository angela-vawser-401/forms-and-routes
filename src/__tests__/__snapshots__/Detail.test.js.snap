// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail component renders 1`] = `
<Fragment>
  <div
    className="Detail"
  >
    <h1 />
    <img />
    <div>
      <p>
        Gender: 
      </p>
      <p>
        Hair: 
      </p>
      <p>
        Affiliation: 
      </p>
    </div>
  </div>
</Fragment>
`;
